### DISCUSSION TITLE
KongHQ Unirest Java with HTTClient5

### DISCUSSION THREAD

**Discussion by hell-storm2004** [Kong/unirest-java#537] [Q&A] [answered]
I just wanted to know , is unirest-java compatible with Apache httpclient5?

**Comment by ryber** [comment, marked as answer]
No, Http Client 5 took a radically different approach from previous versions when it comes to asynchronous requests. They are now fully reactive. That pattern just did not jive with how Unirest does async. I took at least 3 different full attempts to make it fit but in the end chose to move towards the native Java HttpClient introduced in Java 1.9 rather than stay with Apache

**Reply by hell-storm2004** [reply]
Sorry i forgot to reply back. Thanks for the response.
